extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {CSS} from '../../../build/amp-social-share-0.2.css';
+import {Fragment, createElement} from '../../../src/preact';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {SocialShare} from './social-share';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {isExperimentOn} from '../../../src/experiments';
+import {toggle} from '../../../src/style';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-social-share';
+
+class AmpSocialShare extends PreactBaseElement {
+  /** @override */
+  init() {
+    let rendered;
+    const host = this.element;
+    const dataParams = getDataParamsFromAttributes(host);
+    const render = (_, ...children) => {
+      // We only render once in AMP mode, but React mode may rerender
```

Comment:
If there's nothing destructive about our render, then we can support multiple rerenders without issue. DateDisplay destroyed its `<template>` element, which is why I had to prevent rerendering.